libs/@local/graph/sdk/typescript/src/harpc.ts

```diff
@@ -0,0 +1,110 @@
+// This code is written in a way that be *easily* auto generated, which is why we use the classes.
+
+import { ClientError } from "@local/harpc-client";
+import { Decoder, Encoder } from "@local/harpc-client/codec";
+import {
+  Connection,
+  Request,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars -- otherwise TypeScript will fail on inference, I don't know why
```

Comment:
For later – explicitly typing `echo` might help. We probably want to hide the various errors that shouldn't be the caller's concern in any case (`ProcedureIdtooLarge` etc) and roll then up into some 'the SDK messed up' thing, it's a bit noisy at the moment.

Reply:
Tracked in H-3767